Formula/vte3.rb

```diff
@@ -27,18 +26,33 @@ class Vte3 < Formula
   depends_on macos: :mojave
   depends_on "pcre2"
 
+  on_macos do
+    depends_on "llvm" => [:build, :test] if DevelopmentTools.clang_build_version <= 1200
+  end
+
   on_linux do
     depends_on "linux-headers@4.15" => :build
-    depends_on "gcc" # for C++17
+    depends_on "gcc" # for C++20
     depends_on "systemd"
   end
 
-  fails_with gcc: "5"
+  fails_with :clang do
+    build 1200
+    cause "Requires C++20"
+  end
+
+  fails_with :gcc do
+    build "9"
```

Comment:
We've usually been using `version` rather than `build` for GCC. Don't really mind either, but should probably use same style across formulae.

Though, not directly documented in https://docs.brew.sh/Formula-Cookbook#compiler-selection. I guess syntax is somewhat available under hyperlink to https://rubydoc.brew.sh/Formula#fails_with-class_method

Reply:
Yeah `version` makes more sense here. `build` is a bit of an Apple Clang term.

I'm going to bundle this into the icu4c PR. Just need to sort out merge strategy for that first.